src/Illuminate/Bus/DynamoBatchRepository.php

```diff
@@ -0,0 +1,535 @@
+     *
+     * @var string
+     */
+    protected $table;
+
+    /**
+     * The time-to-live value for batch records.
+     *
+     * @var int
+     */
+    protected $ttl;
+
+    /**
+     * The name of the time-to-live attribute for batch records.
+     *
+     * @var string
+     */
+    protected $ttlAttribute;
+
+    /**
+     * The DynamoDB marshaler instance.
+     *
+     * @var \Aws\DynamoDb\Marshaler
+     */
+    protected $marshaler;
+
+    /**
+     * Create a new batch repository instance.
+     */
+    public function __construct(
+        BatchFactory $factory,
+        DynamoDbClient $dynamoDbClient,
+        string $applicationName,
+        string $table,
+        ?int $ttl,
+        ?string $ttlAttribute
+    ) {
+        $this->factory = $factory;
+        $this->dynamoDbClient = $dynamoDbClient;
+        $this->applicationName = $applicationName;
+        $this->table = $table;
+        $this->ttl = $ttl;
+        $this->ttlAttribute = $ttlAttribute;
+        $this->marshaler = new Marshaler;
+    }
+
+    /**
+     * Retrieve a list of batches.
+     *
+     * @param  int  $limit
+     * @param  mixed  $before
+     * @return \Illuminate\Bus\Batch[]
+     */
+    public function get($limit = 50, $before = null)
+    {
+        $condition = 'application = :application';
+
+        if ($before) {
+            $condition = 'application = :application AND id < :id';
+        }
+
+        $result = $this->dynamoDbClient->query([
+            'TableName' => $this->table,
+            'KeyConditionExpression' => $condition,
+            'ExpressionAttributeValues' => array_filter([
+                ':application' => ['S' => $this->applicationName],
+                ':id' => array_filter(['S' => $before]),
+            ]),
+            'Limit' => $limit,
+        ]);
+
+        return array_map(
+            fn ($b) => $this->toBatch($this->marshaler->unmarshalItem($b, mapAsObject: true)),
+            $result['Items']
+        );
+    }
+
+    /**
+     * Retrieve information about an existing batch.
+     *
+     * @param  string  $batchId
+     * @return \Illuminate\Bus\Batch|null
+     */
+    public function find(string $batchId)
+    {
+        if ($batchId === '') {
+            return null;
+        }
+
+        $b = $this->dynamoDbClient->getItem([
+            'TableName' => $this->table,
+            'Key' => [
+                'application' => ['S' => $this->applicationName],
+                'id' => ['S' => $batchId],
+            ],
+        ]);
+
+        if (! isset($b['Item'])) {
+            // If we didn't find it via a standard read, attempt consistent read...
+            $b = $this->dynamoDbClient->getItem([
+                'TableName' => $this->table,
+                'Key' => [
+                    'application' => ['S' => $this->applicationName],
+                    'id' => ['S' => $batchId],
+                ],
+                'ConsistentRead' => true,
```

Comment:
The updates are where it actually matters and those are done atomically where AWS guarantees it updates the right data.
The updates are also independent from the read data because they use update expressions `SET count = count + 1`.
Because of this a `ConsistentRead` isn't required in general.

`ConsistentRead` is also 2x the cost of a read on AWS so I don't think it should be imposed as a default.